### PR TITLE
vim-patch:9.1.0013: Modula2 filetype support lacking

### PR DESCRIPTION
**Pull request by seandewar**
Problem:  Modula2 filetype support lacking
Solution: Improve the Modula-2 runtime support, add additional modula2
          dialects, add compiler plugin, update syntax highlighting,
          include syntax tests, update Makefiles (Doug Kearns)

https://github.com/vim/vim/commit/68a89470693c7687d4e736ca056c05de632e3ac7

- Luaify the detection script: (feel free to roast as I'm still not super familiar with filetype.lua :innocent:)

  - Split the `(*!m2foo*)` and `(*!m2foo+bar*)` detection into two Lua patterns,
    as Lua capture groups cannot be used with `?` and friends (as they only work
    on character classes).

  - Use `vim.api.nvim_buf_call()` (ew) to call `modula2#SetDialect()` to ensure
    `b:modula2` is set for the given bufnr.

- Skip the syntax screendump tests. (A shame as they test some of the detection
  from `(*!m2foo+bar*)` tags, but I tested this locally and it seems to work)

- Port the synmenu.vim changes from Vim9 script. (Also tested this locally)

- (And also add the missing comma for `b:browsefilter` from earlier.)

### CODE REVIEW

File: runtime/lua/vim/filetype/detect.lua [hunk 1]
```diff
@@ -419,6 +419,55 @@ function M.dtrace(_, bufnr)
...
+        fn['modula2#SetDialect'](dialect, extension)
+      end)
+    end
+end
+
+--- @param bufnr integer
+--- @return boolean
+local function is_modula2(bufnr)
+  return matchregex(nextnonblank(bufnr, 1), [[\<MODULE\s\+\w\+\s*;\|^\s*(\*]])
+end
```
Comment (clason) [inline runtime/lua/vim/filetype/detect.lua:457]: could just inline this check?

Comment (seandewar) [reply runtime/lua/vim/filetype/detect.lua:457]: I thought about it, but it's used by both `M.mod` and `M.def`

Comment (clason) [reply runtime/lua/vim/filetype/detect.lua:457]: Perfect reason for having it as a local function.

Comment (zeertzjq) [inline runtime/lua/vim/filetype/detect.lua:457]: It may be better to put this before `modula2()` to match the order in `ft.vim`.

**Review by zeertzjq — APPROVED** [review]

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**